### PR TITLE
ASDPLNG-125: Fix typos/bugs, add update notices, update yum exclude

### PR DESCRIPTION
**Pull request by billglick**
Fix typos in motd notice for 'any/all', weeks, repeated 'the'
    Add timezone to motd notice
    Default week moved to '2'
    Add 1, 24, & 48 hour wall warnings
    Switch update_hour & update_minute parameters to INT for math
    Clean up cron command for dnf-automatic
    Change yum exclude to only kernel
    Change *::update_months parameter names to plural
    Move common class into init
    Move scripts into $root_cron_scripts_dir parameter
    Change yum_cron::exclude to excluded_packages array
    Replace kernel_upgrade::nowait with random_delay that is passed to script as --maxwait parameter
    Rename yum_cron class to yum_update & refactor to move OS specific stuff into hiera
    Fix random_delay if 0 or <=5
    Add specific update_months to motd notice
    Refactor *_upgrade::update_months parameter to simplify logic

This is being tested in asd-pup-control repo with the following hosts:
- asd-pup
- asd-test-afs01
- asd-test-afsdb1
- asd-test-centos7a
- asd-test-rhel8b

### CODE REVIEW

**Review by bsper2 — APPROVED** [review]
Looks good.

**Comment by billglick** [comment]
Only thing I did not address from @andylytical 's recommendations:

> kernel_upgrade.sh line 96: the option --disableexcludes=all could be problematic if certain repos are supposed to be excluded (such as slurm, kubernetes, docker, etc.).

I think this is safe because it is being limited to only update packages that start with `kernel`. It is fairly common to exclude kernel packages in a yum config file (which this module is doing by default), so we need a way to allow this, which is why this script has to disable excludes when it upgrades the kernel package.

File: manifests/kernel_upgrade.pp [hunk 1]
```diff
@@ -127,8 +149,50 @@
...
+                        ), '9', 'Sep',
+                      ), '8', 'Aug'
+                    ), '7', 'Jul'
+                  ), '6', 'Jun'
+                ), '5', 'May'
+              ), '4', 'Apr'
+            ), '3', 'Mar'
+          ), '2', 'Feb'
+        ), '1', 'Jan'
+      )
```
Comment (billglick) [inline manifests/kernel_upgrade.pp:192]: I know this is friggin ugly. It works, but would love to have a more manageable way to convert the $update_months into a pretty string for the motd notice.

Any suggestions for how to improve this?

**Review by andylytical — CHANGES_REQUESTED** [review]
## General
- both files `run-if-true` and `kernel_upgrade` are always installed ... might as well install both from init.pp (ie: `file{ ensure => directory, recurse => true, ...}`. Results in simpler code, thus easier to maintain.
- alternatively, install `kernel_upgade` only if `enabled` is true. (less preferred than above)

## Kernel Upgrade
- As discussed privately, `update_months` should be an array of 3-letter month names. Empty array (default) means every month.
- Update docs to reflect expected input for `update_months`

## Init
- Suggest move `include` statements to the top so they are more visible. Ordering doesn't matter.

File: manifests/kernel_upgrade.pp (hunk 1 above)
Comment (andylytical) [reply manifests/kernel_upgrade.pp:192]: discussed privately and resolution included in PR comment below.

**Comment by andylytical** [comment]
> Only thing I did not address from @andylytical 's recommendations:
> 
> > kernel_upgrade.sh line 96: the option --disableexcludes=all could be problematic if certain repos are supposed to be excluded (such as slurm, kubernetes, docker, etc.).
> 
> I think this is safe here because the updated packages are being limited to only those that start with `kernel`. It is fairly common to exclude kernel packages in a yum config file (which this module is doing by default), so we need a way to allow this, which is why this script has to disable excludes when it upgrades the kernel package.

Sounds good to me.

**Review by andylytical — APPROVED** [review]
Looks good to me! Thanks for all the hard work and continued effort!

**Pull request merged**